core/src/components/input/input.utils.ts

```diff
@@ -0,0 +1,24 @@
+import { printIonError } from '@utils/logging';
+
+export const getCounterText = (
+  inputEl: HTMLIonInputElement,
+  counterFormatter?: (inputLength: number, maxLength: number) => string
+) => {
+  const length = inputEl?.value?.toString().length ?? 0;
```

Comment:
~ Thoughts on checking and returning early if the input element or max length are undefined vs. using the definite assignment assertion multiple time?

OR

Have `getCounterText` accept `value` and `maxlength` as arguments vs. the entire input element.

Reply:
Good thought. I copied this from the original counter implementation, but I like your approach better. I updated this in https://github.com/ionic-team/ionic-framework/pull/26200/commits/613bdcf7ccb8321191948cec5c89f1c480969b62.